omroneipApp/src/drvOmroneip.cpp

```diff
@@ -158,211 +126,260 @@ asynStatus drvOmronEIP::drvUserCreate(asynUser *pasynUser, const char *drvInfo,
+      }
+      else if (keyWords.at("dataType") == "LINT")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt64, &asynIndex);
+        setInteger64Param(asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "USINT")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt32, &asynIndex);
+        setIntegerParam(asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "UINT")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt32, &asynIndex);
+        setIntegerParam(asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "UDINT")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt32, &asynIndex);
+        setIntegerParam(asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "ULINT")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt64, &asynIndex);
+        setInteger64Param(asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "REAL")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamFloat64, &asynIndex);
+        setDoubleParam(asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "LREAL")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamFloat64, &asynIndex);
+        setDoubleParam(asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "STRING")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamOctet, &asynIndex);
+        setStringParam(asynIndex, "");
+      }
+      else if (keyWords.at("dataType") == "WORD")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt8Array, &asynIndex);
+        doCallbacksInt8Array(0, 1, asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "DWORD")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt8Array, &asynIndex);
+        doCallbacksInt8Array(0, 1, asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "LWORD")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt8Array, &asynIndex);
+        doCallbacksInt8Array(0, 1, asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "UDT")
+      {
+        thisAsynStatus = createParam(drvInfo, asynParamInt8Array, &asynIndex);
+        doCallbacksInt8Array(0, 1, asynIndex, 0);
+      }
+      else if (keyWords.at("dataType") == "TIME")
+      {
+        if (newDrvUser->readAsString)
+        {
+          thisAsynStatus = createParam(drvInfo, asynParamOctet, &asynIndex);
+          setStringParam(asynIndex, "");
+        }
+        else
+        {
+          thisAsynStatus = createParam(drvInfo, asynParamInt64, &asynIndex);
+          setInteger64Param(asynIndex, 0);
+        }
+      }
+      else
+      {
+        asynPrint(pasynUserSelf, ASYN_TRACE_ERROR, "%s:%s Err, Invalid datatype: %s\n", driverName, functionName, keyWords.at("dataType").c_str());
+      }
     }
 
-    else if (keyWords.at("dataType") == "UDT")
+    if (thisAsynStatus == asynSuccess)
```

Comment:
I would recommend to turn this around and check for `if (thisAsynStatus != asynSuccess)` here instead because we can keep the actual handling of the error close to the check - and if there is no error everything is good and program flow just continues. It makes for more readable code as well as less indentation for the 'normal' flow of code.

Reply:
Ok, I will change this one to make it a bit clearer. It dont think it will reduce indentation though, unless I use a goto and I know how you feel about them.